### PR TITLE
feat(notifications-request-updates):Notifications for Request Updates

### PR DESCRIPTION
**Pull request by kalisaNkevin**
#### What does this PR do?

`.` make sure that in-app notification and email notification are possible on nomad API

#### Description of Task to be completed?

`.` and the the function which can be used to send the notification to the user
`.` add the notification to some the endpoint which are related to the trip request

#### How should this be manually tested?

` ` to test this function you need to have authenticate socket (manager or requester) to api
`.` when the appropriate action is done he should receive the notification depending on his preference

#### What are the relevant pivotal tracker stories?
[#182201586](https://www.pivotaltracker.com/story/show/182201586)

### CODE REVIEW

**Review by ishimweeli — DISMISSED** [review]

**Review by munyanezaarmel — DISMISSED** [review]

**Review by munyanezaarmel — DISMISSED** [review]

**Review by ishimweeli — DISMISSED** [review]

**Review by Francoise02 — DISMISSED** [review]

**Review by uwishema11 — DISMISSED** [review]

File: src/controllers/approveOrRejectRequest.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import * as ApplicationError from '../utils/errors/applicatioErrors';
 import models from '../database/models';
+import * as notification from '../services/notificationService';
+// import { sendStatusNotification } from '../utils/helpers/sendEmailNotification';
```
Comment (ericus123) [inline src/controllers/approveOrRejectRequest.js]: remove unused code

File: src/controllers/notificationController.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import {
+  addNotificationsStatus,
+  allNotifications,
+  checkIfBlocked,
+  getById,
+} from '../services/notificationService';
+import { internalServerError, notFoundError } from '../utils/errors/applicatioErrors';
+import { successResponse } from '../utils/response';
+// import * as notFound from '../utils/errors/notFoundError';
```
Comment (ericus123) [inline src/controllers/notificationController.js]: remove unused code

File: src/controllers/user.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import { User } from '../database/models';
+import errorResponse from '../utils/error';
+import successResponse from '../utils/success';
+import imageUploader from '../helpers/photoUpload';
+exports.updateUserProfile = async (req, res) => {
+  const user = req.user.dataValues;
+  // console.log(req.user)
```
Comment (ericus123) [inline src/controllers/user.js]: remove unused code

**Review by Francoise02 — DISMISSED** [review]

**Review by munyanezaarmel — DISMISSED** [review]

**Review by ishimweeli — DISMISSED** [review]

**Review by ericus123 — DISMISSED** [review]

File: src/controllers/approveOrRejectRequest.js (hunk 1 above)
Comment (kalisaNkevin) [reply src/controllers/approveOrRejectRequest.js]: changed.

File: src/controllers/user.js (hunk 3 above)
Comment (kalisaNkevin) [reply src/controllers/user.js]: changed

File: src/controllers/notificationController.js (hunk 2 above)
Comment (kalisaNkevin) [reply src/controllers/notificationController.js]: changed

**Review by munyanezaarmel — APPROVED** [review]

**Review by uwishema11 — APPROVED** [review]

**Review by Julienish14 — APPROVED** [review]

**Review by ishimweeli — APPROVED** [review]

**Review by Francoise02 — APPROVED** [review]

**Review by ericus123 — APPROVED** [review]

**Pull request merged**